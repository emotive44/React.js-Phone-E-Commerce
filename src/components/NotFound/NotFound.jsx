import React from 'react';
import ButtonWrapper from '../styles/Button'
import ButtonContainer from '../styles/Button';
import { Link } from 'react-router-dom';

class NotFound extends React.Component {
    render() {
        return(
            <div className="container">
                <div className="row">
                    <div className="col-10 mx-auto text-center text-capitalize pt-5">
                        <h1 className="display-3">404</h1>
                        <h2>Error</h2>
                        <h2>Page not found</h2>
                        <h3>
                            teh requested URL &nbsp;
                            <span className="text-danger">
                                {this.props.location.pathname}
                            </span>
                            &nbsp; was not found
                        </h3>
                        <ButtonContainer>
                            <Link to='/' style={{"text-decoration": "none"}}>
                                {'<-- Go Back To Home Page'}
                            </Link>
                        </ButtonContainer>
                    </div>
                </div>
            </div>
        )
    }
}

export default NotFound;