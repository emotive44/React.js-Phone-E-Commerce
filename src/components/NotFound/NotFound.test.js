import React from 'react';
import NotFound from './NotFound';
import renderer from 'react-test-renderer';

jest.mock('react-router-dom', () => ({
    Link: 'Link'
}));

describe('NotFound Component', () => {
    test('should render return null if pathname is undefiend', () => {
        const component = renderer.create(<NotFound />);
        expect(component.toJSON()).toMatchSnapshot();
    })
})