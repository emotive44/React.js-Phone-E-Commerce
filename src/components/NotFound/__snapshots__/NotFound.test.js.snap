// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Component should render return null if pathname is undefiend 1`] = `null`;
